victorLi/assignments/treesDFS/lc113/lc113.js

```diff
@@ -22,18 +37,67 @@ class TreeNode {
 
 
 const find_paths = function(root, sum) {
-  allPaths = [];
-  // TODO: Write your code here
-  return allPaths;
-};
+    allPaths = [];
+    // TODO: Write your code here
+    let prevSum = 0;
+    let stack = []
+    let sumStack = []
+    let onePath = []
+    
+    let node = root;
 
+    while( node || stack.length )
+    {
+      while( node )
+      {
+          stack.push(node)
+          onePath.push(node.value)
+          sumStack.push(prevSum += node.value)
+          node = node.left
+      }
+
+      
+      node = stack.pop()
+
+    
+    
+      if( isLeaf(node) && prevSum === sum)
+      {
+          allPaths.push(onePath)
+          console.log("-----ONEPATH--------------")
+          console.log(onePath)
+     
+          
+      }
+      
+
+      if( onePath.length > 1 &&  !node.right)
+         onePath.pop()
+      
+  
+      
+      node = node.right;
+      prevSum = sumStack.pop()
+
+    }
+
+    return allPaths;
+};
 
+let isLeaf = (node) =>{
+  if( !node.right && !node.left )
+    return true
+  else
+    return false
+}
 
 var root = new TreeNode(12)
 root.left = new TreeNode(7)
 root.right = new TreeNode(1)
 root.left.left = new TreeNode(4)
 root.right.left = new TreeNode(10)
 root.right.right = new TreeNode(5)
+root.right.right.left = new TreeNode(5)
+root.right.right.right = new TreeNode(3)
```

Comment:
Nice job adding a custom test case, but make sure one of the test cases you make checks the edge case(s) you list beforehand.